libs/src/sdk/middleware/addAppNameMiddleware.ts

```diff
@@ -1,27 +1,58 @@
+import { DeveloperAppsApi } from '../api/developer-apps/DeveloperAppsApi'
 import {
   type Middleware,
   type RequestContext,
   type FetchParams,
+  Configuration,
   querystring
 } from '../api/generated/default'
 
+let appName: string | undefined
+
 /**
  * Appends the configured app_name to the query string for tracking API usage
  * @param options the middleware options
  * @param {string} options.appName the name of the app using the SDK
  */
 export const addAppNameMiddleware = ({
-  appName
+  appName: providedAppName,
+  services
 }: {
-  appName: string
+  appName?: string
+  services: any
 }): Middleware => {
+  appName = providedAppName
   return {
-    pre: async (context: RequestContext): Promise<FetchParams> => ({
-      url:
-        context.url +
-        (context.url.includes('?') ? '&' : '?') +
-        querystring({ app_name: appName }),
-      init: context.init ?? {}
-    })
+    pre: async (context: RequestContext): Promise<FetchParams> => {
+      // If an app name is not provided, fetch the name from the dev app
+      if (!appName) {
+        const discOnlyMiddleware = [
+          services.discoveryNodeSelector.createMiddleware()
+        ]
+        const generatedApiClientConfigSansAppName = new Configuration({
```

Comment:
These names were relevant when we had this stuff in the main sdk init but they can be less specific now. Maybe just `middleware`, `apiConfig` and `developerApps`